### PR TITLE
Update Fleet.DiagnosticData to 2.0.0

### LINKED ISSUE #418: [Model Update]: bamm:io.catenax.fleet.diagnostic_data:1.0.0 to version 2.0.0 with SAMM
## Update Reason
The use case quality group decided to upgrade this model to version 2.0.0: 
- Use new versions of ESMF toolings
- Convert from BAMM to SAMM
- Make adaptions in model structure

## MS1 Criteria

- [X] The model that should be updated exists
- [X ] The referenced use case exists: It is use case CX QAX
- [X] The potential updates are discussed with all stakeholders
- [X] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
--> new version will be 2.0.0
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [X] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by TorstenHirsch**
## Description
This PR reflects all changes for Fleet.DIagnosticData 2.0.0:
### Added
- metaInformation property added to root entity
- integration of the shared UUID characteristic and RegEx for Catena-X identifiers
- introducing new characteristic for unique identifiers
- start and end date of a session
- procedures to support failure-tree-based analysis in the workshop: Procedures can have sub-procedures could also be linked to ecuList
- for Ecus: calibration and variant coding properties

### Changed
- changed all properties to optional except unique identifier per entity
- Clear structure: session has ecuList as child, ecuList has dtcList as child, dtcList has envConditionList as child

### Removed
- vehicle entity: Properties now located one level up on DiagnosticSession
- eventList: Removed completely - instead procedures list was introduced

Closes #418 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by TorstenHirsch** [comment]
@agg3fe please do the review - the most complex model of QAX
modeling_team

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 1]
```diff
@@ -0,0 +1,541 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:1]: please add copyright headers

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+@prefix : <urn:samm:io.catenax.fleet.diagnostic_data:2.0.0#> .
+@prefix ext-uuid: <urn:samm:io.catenax.shared.uuid:1.0.0#> .
+
+:DiagnosticData a samm:Aspect ;
+   samm:description "A Catena-X semantic data model for a list of vehicle diagnostic sessions."@en ;
+   samm:properties ( :diagnosticSessions [ samm:property :metaInformation; samm:optional true ] ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
+
+:diagnosticSessions a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:17]: Property and the referenced Characteristic should not have the same name

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "Pontiac Car Diagnostic software" .
+
+:diagnosticSoftwareVersion a samm:Property ;
+   samm:preferredName "Diagnostic software version"@en ;
+   samm:description "The version of the used diagnostic software"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "V.411 Patch 2023-02" .
+
+:mileage a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:95]: Property and the referenced Characteristic should not have the same name

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+   samm:preferredName "Workshop"@en ;
+   samm:description "Groups workshop-related properties"@en ;
+   samm:characteristic :WorkshopCharacteristic .
+
+:ecuList a samm:Property ;
+   samm:preferredName "ECU list"@en ;
+   samm:description "list of ECUs that had an entry in its internal failure memory during the diagnostic session"@en ;
+   samm:characteristic :Ecus .
+
+:procedures a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:135]: Property and the referenced Characteristic should not have the same name

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "0001" .
+
+:variantCoding a samm:Property ;
+   samm:preferredName "Variant coding"@en ;
+   samm:description "Variant coding that was put on this ecu. Variant coding tells the ecus software how to behave."@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "Variant: No trailer." .
+
+:dtcList a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:310]: `preferredName` missing

Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:310]: Property and the referenced Characteristic should not have the same name

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 6]
```diff
@@ -0,0 +1,541 @@
...
+   samm:description "Characteristic for procedure results."@en ;
+   samm:dataType xsd:string ;
+   samm-c:values ( "ok" "nok" "failed" "null" ) .
+
+:SubProceduresList a samm-c:List ;
+   samm:preferredName "List of sub-procedures"@en ;
+   samm:description "A list of sub-procedures"@en ;
+   samm:dataType :ProcedureCall .
+
+:Latitude a samm:Characteristic ;
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:372]: `preferredName` missing

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+   samm:description "A list of environment conditions: E.g. outside temperature measured by the vehicle, a specific value measured by on ECU, ...."@en ;
+   samm:characteristic :EnvironmentConditions .
+
+:HexValueTraitNullable a samm-c:Trait ;
+   samm:preferredName "Hex Value Trait Nullable"@en ;
+   samm:description "This trait allows 0-9, a-f, A-F in a string value. Optional prefix is \"0x\"."@en ;
+   samm-c:baseCharacteristic :HexCodeCharacteristic ;
+   samm-c:constraint :HexRegularExpressionNullable .
+
+:Long a samm:Characteristic ;
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:480]: `preferredName` and `description` missing

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl (hunk 2 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:37]: fixed with next upload

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl (hunk 3 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:115]: fixed with next upload

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl (hunk 5 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:332]: fixed with next upload

Comment (TorstenHirsch) [reply io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:332]: fixed with next upload

**Comment by TorstenHirsch** [comment]
@agg3fe Please reassess - I hope I covered every finding
modeling_team

File: io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl [hunk 8]
```diff
@@ -0,0 +1,570 @@
+#######################################################################
+# Copyright (c) 2022, 2024 Robert Bosch GmbH
+# Copyright (c) 2022, 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+# Copyright (c) 2022, 2024 Volkswagen AG
+# Copyright (c) 2022, 2024 ZF Friedrichshafen AG
+# Copyright (c) 2022, 2024 SAP SE
+# Copyright (c) 2022, 2024 Siemens AG
+# Copyright (c) 2022, 2024 Contributors to the Eclipse Foundation
+# Copyright (c) 2022 Mercedes Benz AG
```
Comment (agg3fe) [inline io.catenax.fleet.diagnostic_data/2.0.0/DiagnosticData.ttl:9]: missing 2024 entry here?

**Pull request merged**